### DISCUSSION TITLE
[Feature] iOS App Actively Queries DNS SVCB/HTTPS Records

### DISCUSSION THREAD

**Discussion by Jackie264** [immich-app/immich#14938] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Due to the home broadband being on a NAT3 network, it is necessary to use STUN technology to obtain the public IP address and port number. However, the port number changes frequently. We use SVCB/HTTPS records provided by the DNS provider to redirect access to https://immich.example.com to https://immich.example.com:port. This approach works well in browsers (Safari/Chrome/Firefox), but on iOS apps, the port number must be manually entered; otherwise, the app cannot connect to the server.

```
root@ThinkPad-X1:~# dig immich.example.com https

; <<>> DiG 9.18.28-0ubuntu0.24.04.1-Ubuntu <<>> immich.example.com https
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 26252
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;immich.example.com.            IN      HTTPS

;; ANSWER SECTION:
immich.example.com.     10      IN      CNAME   svc.example.com.
svc.example.com.       10      IN      HTTPS   1 . alpn="h2,http/1.1" port=22351

;; Query time: 4 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Thu Dec 26 10:46:58 CST 2024
;; MSG SIZE  rcvd: 102
```

I believe there are many users in such scenarios. If iOS apps could actively query DNS SVCB/HTTPS records, this issue could be perfectly resolved.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
> I believe there are many users in such scenarios.

To set expectations: I doubt this, as I've never seen anybody describe the scenario you're in. I don't think this is something the team will be spending time on, but resolving these record types seems reasonable so a PR would be welcomed.

**Comment by bo0tzz** [comment]
Closing, see https://github.com/immich-app/immich/pull/15029#issuecomment-2695118755
